src/components/application_manager/src/commands/mobile/set_app_icon_request.cc

```diff
@@ -253,6 +249,10 @@ void SetAppIconRequest::on_event(const event_engine::Event& event) {
         const std::string& path =
             (*message_)[strings::msg_params][strings::sync_file_name]
                        [strings::value].asString();
+        if (is_icons_saving_enabled_) {
```

Comment:
@VVeremjova The result checked above in line 240 is the result that will be sent to the mobile. Maybe you should check the content of response_info variable